maven-resolver-provider/src/site/apt/dependency-types.apt

```diff
@@ -0,0 +1,74 @@
+|| type                 || classifier   || extension || language || path types           || includesDependencies ||
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+|| Maven                ||              ||           ||          ||                      ||                      ||
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+| <<<pom>>>             |               | <= type>   | none      |                       |                       |
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+| <<<bom>>> *           |               | <<<pom>>>  | none      |                       |                       |
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+| <<<maven-plugin>>>    |               | <<<jar>>>  | java      | classes               |                       |
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+|| Java                 ||              ||           ||          ||                      ||                      ||
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+| <<<jar>>>             |               | <= type>   | java      | classes, modules      |                       |
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+| <<<java-source>>>     | <<<sources>>> | <<<jar>>>  | java      |                       |                       |
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+| <<<javadoc>>>         | <<<javadoc>>> | <<<jar>>>  | java      | classes               |                       |
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+| <<<test-jar>>>        | <<<tests>>>   | <<<jar>>>  | java      | classes, patch module |                       |
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+| <<<modular-jar>>> *   | <<<tests>>>   | <<<jar>>>  | java      | modules               |                       |
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+| <<<classpath-jar>>> * | <<<tests>>>   | <<<jar>>>  | java      | classes               |                       |
+*-----------------------+---------------+------------+-----------+-----------------------+-----------------------+
+| <<<fatjar>>> *        | <<<tests>>>   | <<<jar>>>  | java      | classes               |                       |
```

Comment:
fatjar == jar, but:
* can be on classes only (never on modules path)
* stops resolution (just like any existing self sufficient type, i.e. war)

Reply:
@cstamas what does this mean in term of this PR?
is there something wrong
is there something to add?